tests/test_parameter.py

```diff
@@ -126,6 +127,15 @@ def test_parameter_set_value(parameter):
     assert_parameter_attributes(par, changed_attribute_values)
 
 
+def test_parameter_change_value():
+    """Test if value can be set to a value that was previous out of bounds"""
+    p = lmfit.Parameter("test", value=5, min=4, max=10)
+    assert_paramval(p, 5, tol=1.0e-2)
+
+    p.set(value=22, min=20, max=30)
+    assert_paramval(p, 22, tol=1.0e-2)
+
```

Comment:
I think this test should ideally be incorporated in the one above (`test_parameters_set_value`) with a one line comment and reference to the GitHub Issue. I'd rather not start using `assert_paramval` here (as I've been trying to remove the use of that function); instead please use the `assert_parameter_attributes` as in the surrounding tests, so that all other attributes get checked as well.

Reply:
just pushed an update. Let me know if other changes are needed. Thanks